extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/AvroAttributeRowBuilder.java

```diff
@@ -0,0 +1,48 @@
+package mil.nga.giat.geowave.adapter.vector;
+
+import java.util.HashMap;
+
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.index.StringUtils;
+import mil.nga.giat.geowave.core.store.adapter.NativeFieldHandler.RowBuilder;
+import mil.nga.giat.geowave.core.store.data.PersistentValue;
+
+import org.opengis.feature.simple.SimpleFeature;
+
+/**
+ * A GeoWave RowBuilder, used internally by AbstractDataAdapter to construct
+ * rows from a set field values (in this case SimpleFeatures from a set of
+ * attribute values). This implementation simply wraps a geotools
+ * SimpleFeatureBuilder.
+ * 
+ */
+public class AvroAttributeRowBuilder implements
+		RowBuilder<SimpleFeature, Object>
+{
+	private final HashMap<String, Object> idToValue = new HashMap<String, Object>();
```

Comment:
Does this need to be a hashmap?  We're only expecting a single fieldValue to be set (the one which represents the entire serialized Simple Feature).